src/System.Private.Uri/src/System/UriExt.cs

```diff
@@ -559,10 +560,19 @@ public static string UnescapeDataString(string stringToUnescape)
 
                     UnescapeMode unescapeMode = UnescapeMode.Unescape | UnescapeMode.UnescapeAll;
                     position = 0;
-                    char[] dest = new char[stringToUnescape.Length];
-                    dest = UriHelper.UnescapeString(stringToUnescape, 0, stringToUnescape.Length, dest, ref position,
+                    ValueStringBuilder pooledArray = new ValueStringBuilder(stringToUnescape.Length);
```

Comment:
Nit: `pooledArray` as a name makes it sound like this a `T[]`.  Also whether VSB uses a pool is an implementation detail.  Let's just name this `vsb` or `result` or `dest` or some such thing.